src/kessel/console/__init__.py

```
@@ -0,0 +1,100 @@
+    try:
+        decoded = json.loads(b64decode(header))
+    except Exception as exc:
+        raise ValueError(f"Failed to decode identity header: {exc}") from exc
+
+    if not isinstance(decoded, dict):
+        raise ValueError("Identity header did not decode to a JSON object")
+
+    try:
+        identity = decoded["identity"]
+    except KeyError:
+        raise ValueError("Identity header is missing the 'identity' envelope key")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)